src/datachain/catalog/catalog.py

```diff
@@ -1198,40 +1252,77 @@ def get_remote_dataset(
 
     def get_dataset_dependencies(
         self,
-        name: str,
-        version: str,
+        name: str | None = None,
+        version: str | None = None,
         namespace_name: str | None = None,
         project_name: str | None = None,
         indirect=False,
+        dataset_id: int | None = None,
+        dataset_version_id: int | None = None,
     ) -> list[DatasetDependency | None]:
-        dataset = self.get_dataset(
-            name,
-            namespace_name=namespace_name,
-            project_name=project_name,
-        )
+        if (name and dataset_id) or (version and dataset_version_id):
+            raise ValueError(
+                "Invalid arguments: only one set of identifiers should be supplied. "
+                "Provide either (name and version),(dataset_id and dataset_version_id)"
+            )
 
-        direct_dependencies = self.metastore.get_direct_dataset_dependencies(
-            dataset, version
+        if not dataset_id:
+            if not name or not version:
+                raise ValueError(
+                    "name and version must be provided when dataset_id is not provided"
+                )
+            dataset = self.get_dataset(
+                name,
+                namespace_name=namespace_name,
+                project_name=project_name,
+            )
+            dataset_version = dataset.get_version(version)
+            dataset_id = dataset.id
+            dataset_version_id = dataset_version.id
+
+        if dataset_version_id is None:
+            raise ValueError(
+                "dataset_version_id must be provided when dataset_id is provided"
+            )
+
+        dependency_ids = self.get_dataset_dependency_ids(
+            dataset_id,
+            dataset_version_id,
+            indirect=indirect,
         )
+        all_ids = get_all_ids(dependency_ids)
 
-        if not indirect:
-            return direct_dependencies
+        dependencies = {
+            str(d.id): d
+            for d in self.metastore.get_direct_dataset_dependencies_by_ids(all_ids)
+        }
+        return build_nested_dependencies(dependencies, dependency_ids, indirect)
 
+    def get_dataset_dependency_ids(
+        self,
+        dataset_id: int,
+        version_id: int,
+        indirect=False,
+    ) -> dict[str, dict]:
```

Comment:
Method name and return type are not correlated. `get_dataset_dependency_ids_tree` or something like that? 🤔